internal/generators/azure/azure.go

```
@@ -250,6 +253,20 @@ func (g *AzureOpenAI) Name() string {
 	return "azure.AzureOpenAI"
 }
 
+// SupportsVision reports vision capability based on the active model family.
+// Chat-completion models go through the multipart image_url path in
+// openaicompat.ConversationToMessages, but legacy completion models
+// (gpt-3.5-turbo-instruct, davinci-002, babbage-002, etc.) use the
+// CompletionRequest path that only carries the flattened LastPrompt() text.
+// Reporting vision capability for completion models would silently drop image
+// attachments. See types.VisionCapable.
+func (g *AzureOpenAI) SupportsVision() bool {
+	if completionModels[g.model] {
```

golangci-lint (2.12.2)

[error] S1008: should use 'return !completionModels[g.model]' instead of 'if completionModels[g.model] { return false }; return true'

(staticcheck)